scripts/ai/AIDriveStrategyChopperCourse.lua

```diff
@@ -0,0 +1,782 @@
+            self:debug('Fruitwaypoint was set to the middle of the row %d', fruitCheckWaypoint)
+        else
+            fruitCheckWaypoint = fruitCheckWaypoint + 10
+            self:debug('Fruitwaypoint was set 10 ahead to couldn\'t determine row length %d', fruitCheckWaypoint)
+        end
+    end
+
+    -- If we are on the first headland engage chase mode and the pipeoffset to zero disabled for sugarcane
+    if self.course:isOnHeadland(fruitCheckWaypoint, 1 ) and not self.isSugarCaneHarvester then
+        self:debug('I am on a headland enganing chase mode')
+        self.pipeOffsetX = 0
+        self:setPipeOffsetZ(-self.measuredBackDistance - 2) 
+        -- We need this so the unloader knows what turn manuvers to use
+        self.chaseMode = true
+    else
+        -- Check to see if the waypoint that we are using would put the unloader in the fruit using out current pipe offset
+        -- And if would use the oppisite vaule of the pipe offset as the should be cleared because the should be our last row worked
+        local hasFruit = self:isPipeInFruitAtWaypointNow(self.course, fruitCheckWaypoint, self.pipeOffsetX)
+        self:debug('I found fruit %s at waypoint %d', tostring(hasFruit), fruitCheckWaypoint)
+        if hasFruit then
+            self:debug('I found fruit on my current side switch to the opposite side')
+            self.pipeOffsetX = -self.pipeOffsetX
+            -- Perform another check to see with out updated pipeoffset to make sure that was our last row worked and it is inded clear of fruit
+            -- If we find fruit again that means we no clear lane for the unloader drive and have it follow us down the row we are clearing
+            hasFruit = self:isPipeInFruitAtWaypointNow(self.course, fruitCheckWaypoint, self.pipeOffsetX)
+            if hasFruit and not self.isSugarCaneHarvester then
+                self:debug('I found fruit again I must be on a land row switch to chase mode')
+                self.pipeOffsetX = 0
+                self:setPipeOffsetZ(-self.measuredBackDistance - 2)
+                -- We need this so the unloader knows what turn manuvers to use
+                self.chaseMode = true
+                self.landRow = true
+                return
+            end
+        end
+        
+        -- Once final check to make sure the pipeoffset we are going to supply to the unloader is on the field.
+        local x, _, z = localToWorld(self.storage.fruitCheckHelperWpNode.node, self.pipeOffsetX, 0, 0)
+        local fieldPolygon = self.course:getFieldPolygon()
+        if not CpMathUtil.isPointInPolygon(fieldPolygon, x, z) then
+            -- The point we are using isn't on the field check the otherside of the chopper to see if there is fruit and if there is no fruit use have the unloader follow us on that side
+            -- Otherwise just have the unloader follow us
+            if hasFruit and not self.isSugarCaneHarvester then 

```

Comment:
Also, what case ist this even? Not in the field but there is fruit? On an adjacent field or what?